sklearn/ensemble/_hist_gradient_boosting/loss.py

```diff
@@ -269,6 +273,62 @@ def update_leaves_values(self, grower, y_true, raw_predictions,
             # Note that the regularization is ignored here
 
 
+class Poisson(BaseLoss):
+    """Poisson deviance loss with log-link, for regression.
+
+    For a given sample x_i, Poisson deviance loss is defined as::
+
+        loss(x_i) = y_true_i * log(y_true_i/exp(raw_pred_i))
+                    - y_true_i + exp(raw_pred_i))
+
+    This actually computes half the Poisson deviance to simplify
+    the computation of the gradients.
+    """
+
+    def __init__(self, sample_weight):
+        # If sample weights are provided, the hessians and gradients
+        # are multiplied by sample_weight, which means the hessians are
+        # equal to sample weights.
+        super().__init__(hessians_are_constant=False)
+
+    inverse_link_function = staticmethod(np.exp)
+
+    def pointwise_loss(self, y_true, raw_predictions):
+        # shape (1, n_samples) --> (n_samples,). reshape(-1) is more likely to
+        # return a view.
+        raw_predictions = raw_predictions.reshape(-1)
+        # TODO: For speed, we could remove the constant xlogy(y_true, y_true)
+        # Advantage of this form: minimum of zero at raw_predictions = y_true.
+        loss = (xlogy(y_true, y_true) - y_true * (raw_predictions + 1)
+                + np.exp(raw_predictions))
```

Comment:
Should there be a .5 somewhere?
(I assume not, otherwise the gradient checks would fail, just want to make sure)

Reply:
https://github.com/scikit-learn/scikit-learn/blob/2a5d40f5d102d19fd874134a2443ec0cdc7e0cdc/sklearn/_loss/glm_distribution.py#L314-L315
`1/2 * poisson_deviance = 1/2 * 2 * (xlog(y_true, y_true/y_pred) - y_true + y_pred = xlogy(y_true, y_true) - y_true*log(exp(raw)) - y_true + exp(raw)`
Pretty sure:+1: